### PR TITLE
AppVeyor fixes

### LINKED ISSUE #112: Add support for Go 1.11
At least Windows builds are broken:

```sh
oader.c:19:41: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'struct evmc_instance * (*)(void)' [-Wcast-function-type]
 #define DLL_GET_CREATE_FN(handle, name) (evmc_create_fn) GetProcAddress(handle, name)
                                         ^
loader.c:103:17: note: in expansion of macro 'DLL_GET_CREATE_FN'
     create_fn = DLL_GET_CREATE_FN(handle, name);
                 ^~~~~~~~~~~~~~~~~
loader.c:19:41: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'struct evmc_instance * (*)(void)' [-Wcast-function-type]
 #define DLL_GET_CREATE_FN(handle, name) (evmc_create_fn) GetProcAddress(handle, name)
                                         ^
loader.c:112:25: note: in expansion of macro 'DLL_GET_CREATE_FN'
             create_fn = DLL_GET_CREATE_FN(handle, name);
                         ^~~~~~~~~~~~~~~~~
loader.c:19:41: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'struct evmc_instance * (*)(void)' [-Wcast-function-type]
 #define DLL_GET_CREATE_FN(handle, name) (evmc_create_fn) GetProcAddress(handle, name)
                                         ^
loader.c:117:21: note: in expansion of macro 'DLL_GET_CREATE_FN'
         create_fn = DLL_GET_CREATE_FN(handle, "evmc_create");
                     ^~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by chfast**
Two issues with AppVeyor:
- Go has been upgraded to 1.11, this PR fixes #112.
- AppVeyor cache for Hunter stopped working, bump Hunter version to refresh it.
- Disable dev warnings in CMake.

### CODE REVIEW

File: lib/loader/loader.c [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #define DLL_HANDLE HMODULE
 #define DLL_OPEN(filename) LoadLibrary(filename)
 #define DLL_CLOSE(handle) FreeLibrary(handle)
-#define DLL_GET_CREATE_FN(handle, name) (evmc_create_fn) GetProcAddress(handle, name)
+#define DLL_GET_CREATE_FN(handle, name) (evmc_create_fn)(uintptr_t) GetProcAddress(handle, name)
```
Comment (axic) [inline lib/loader/loader.c]: Doesn't this fix go 1.11 ?

**Comment by axic** [comment]
Where is the Go 1.10 change?

**Comment by chfast** [comment]
The description was not up-to-date. Fixing Go was easier than enforcing Go 1.10.

File: appveyor.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ before_build:
     if ($env:GENERATOR) {
       if (!(test-path build)) { mkdir build }
       cd build
-      cmake -G "$env:GENERATOR" .. -DBUILD_SHARED_LIBS=ON -DEVMC_EXAMPLES=ON -DEVMC_TESTING=ON -DCMAKE_INSTALL_PREFIX=C:\install
+      cmake -Wno-dev -G "$env:GENERATOR" .. -DBUILD_SHARED_LIBS=ON -DEVMC_EXAMPLES=ON -DEVMC_TESTING=ON -DCMAKE_INSTALL_PREFIX=C:\install
```
Comment (axic) [inline appveyor.yml]: Is this `-Wno-dev` supressing the "cmake stack trace" messages?

Comment (chfast) [reply appveyor.yml]: It suppresses warnings like this: https://ci.appveyor.com/project/chfast/evmc/build/243/job/a7d7byb4os8cvm3c
Apparently PowerShell feaks out when it sees any output to standard error.

**Comment by chfast** [comment]
Can I merge this?

**Pull request merged**